Add custom well-known (#13035)
matrix-org/synapse: docs/usage/configuration/config_documentation.md

serve_server_wellknown: true
```
---

> Config option: `custom_wellknown`

Reviewer: ```suggestion
Config option: `custom_well_known`
```

> Config option: `custom_well_known`

Reviewer 2: Bit of a nitpick while passing by and skimming through this PR, I would rename this config option into something like `extra_well_known_content`. As it stands it's not clear to me from the option's name that this option is not replacing the content that's automatically generated.

> This option allows the server to define arbitrary payload in addition to the default wellknown.

Reviewer: ```suggestion
This option allows server runners to add arbitrary key-value pairsto the [client-facing `.well-known` response](https://spec.matrix.org/latest/client-server-api/#well-known-uri).
```

> This option allows server runners to add arbitrary key-value pairsto the [client-facing `.well-known` response](https://spec.matrix.org/latest/client-server-api/#well-known-uri).

Reviewer 2: ```suggestion
This option allows server runners to add arbitrary key-value pairs to the [client-facing `.well-known` response](https://spec.matrix.org/latest/client-server-api/#well-known-uri).
```

If this option is provided, it parses the given yaml to json and 
serves it on `/.well-known/matrix/client` endpoint
alongside the standard properties.

Example configuration:
```yaml
extra_well_known_content: 
  option1: value1
  option2: value2
```

> ---

Reviewer: Can we rename this to `extra_well_known_client_content` please? (To distinguish from /.well-known/server)

Also, can we add a sentence like this to the first paragraph?

```
Note that the `public_baseurl` config option must be provided  for a response to serve `/.well-known/client` at all.
```
